### PR TITLE
Update transaction related bbes

### PR DESCRIPTION
**Pull request by pcnfernando**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/check-semantics/check_semantics.bal [hunk 1]
```diff
@@ -2,9 +2,7 @@ import ballerina/io;
 
 public function main() returns error? {
     do {
-        // If either `foo()` or `bar()` invocations returns an error,
-        // the error will be returned from the `main` function and execution
-        // of the `main` function ends.
+        // If check gets an error from either `foo()` or `bar()` invocations, the error will caught at the `on fail`
```
Comment (praneesha) [inline examples/check-semantics/check_semantics.bal:5]:
```suggestion
        // If  `check` gets an error from either `foo()` or `bar()` invocations, the error will be caught at the `on fail`
```

File: examples/check-semantics/check_semantics.md [hunk 2]
```diff
@@ -1,13 +1,11 @@
 # Check semantics
 
 `check` semantics is not to simply return on an `error` value.
```
Comment (praneesha) [inline examples/check-semantics/check_semantics.md:3]:
```suggestion
`check` semantics does not simply return an `error` value.
```

File: examples/check-semantics/check_semantics.md [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-When `check` gets an `error` value, it fails.
-The enclosing block decides how to handle the failure.
-Most blocks pass the failure up to the enclosing block.
-Function definition handles the failure by returning the error.
+When `check` gets an `error` value, it fails. The enclosing block decides how to handle the failure.
+Most blocks pass the failure up to the enclosing block. Function definition handles the failure by returning the error.
```
Comment (praneesha) [inline examples/check-semantics/check_semantics.md:5]:
```suggestion
Most blocks pass the failure up to the enclosing block. The function definition handles the failure by returning the error.
```

File: examples/check-semantics/check_semantics.md [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 `on fail` can catch the error.
-`fail` statement is like `check`, but it always fails.
-Differs from exceptions in that control flow is explicit.
+
+`fail` statement is like `check`, but it always fails and differs from exceptions in that control flow is explicit.
```
Comment (praneesha) [inline examples/check-semantics/check_semantics.md:8]:
```suggestion
The `fail` statement is like `check`, but it always fails. It differs from the exceptions in that the control flow, which are explicit.
```
Please check if the grammatical changes suggested are affecting the technicalities.

File: examples/retry-transaction-statement/retry_transaction_statement.bal [hunk 5]
```diff
@@ -23,5 +23,4 @@ function doStage2() returns error? {
     // Returns `error:Retriable` error for retrying.
```
Comment (praneesha) [inline examples/retry-transaction-statement/retry_transaction_statement.bal:23]:
```suggestion
    // Returns an `error:Retriable` error after retrying.
```

File: examples/retry-transaction-statement/retry_transaction_statement.md [hunk 6]
```diff
@@ -1,10 +1,9 @@
 # Retry transaction statement
 
 Transactional errors are often transient: retrying will fix them.
-This works by creating a  RetryManager object `r`, before executing the transaction.
-If the block fails with error `e`, it calls `r.shouldRetry(e)`.
-If that returns `true`, then it executes the block again.
-`retry` has an optional type parameter giving class of `RetryManager` to create, and optional arguments to new `DefaultRetryManager` tries `n` times.
+This works by creating a `RetryManager` object `r`, before executing the transaction.
```
Comment (praneesha) [inline examples/retry-transaction-statement/retry_transaction_statement.md:4]:
```suggestion
This works by creating a `RetryManager` object `r` before executing the transaction.
```

File: examples/retry-transaction-statement/retry_transaction_statement.md [hunk 7]
```diff
@@ -1,10 +1,9 @@
...
+If the block fails with error `e`, it calls `r.shouldRetry(e)`. If that returns `true`, then it executes the block again.
```
Comment (praneesha) [inline examples/retry-transaction-statement/retry_transaction_statement.md:5]:
```suggestion
If the block fails with an error `e`, it calls `r.shouldRetry(e)`. If that returns `true`, then it executes the block again.
```

File: examples/retry-transaction-statement/retry_transaction_statement.md [hunk 8]
```diff
@@ -1,10 +1,9 @@
...
+`retry` has an optional type parameter giving class of `RetryManager` to create, and optional arguments to new. `DefaultRetryManager` tries 3 times.
```
Comment (praneesha) [inline examples/retry-transaction-statement/retry_transaction_statement.md:6]:
```suggestion
`retry` has an optional type parameter giving class of `RetryManager` to create, and optional arguments to new. `DefaultRetryManager` tries 3 times.
```
The idea is not clear and is grammatically incorrect. Needs to rewrite.

File: examples/transaction-statement/transaction_statement.bal [hunk 9]
```diff
@@ -1,16 +1,13 @@
 import ballerina/io;
 
 public function main() returns error? {
-    // Compile-time guarantees that transactions are bracketed with
-    // begin and commit or rollback. Transaction statement begins
-    // a new transaction and executes a block.
+    // Transaction statement begins a new transaction and executes a block.
```
Comment (praneesha) [inline examples/transaction-statement/transaction_statement.bal:4]:
```suggestion
    // The transaction statement begins a new transaction and executes a block.
```

File: examples/transaction-statement/transaction_statement.bal [hunk 10]
```diff
@@ -1,16 +1,13 @@
...
     transaction {
         doStage1();
         doStage2();
 
         // Commit of a transaction must be done explicitly using commit.
-        // Commit must be lexically within a transaction statement and may
-        // return an error;
+        // Commit must be lexically within a transaction statement and may return an error;
```
Comment (praneesha) [inline examples/transaction-statement/transaction_statement.bal:10]:
```suggestion
        // A commit must be lexically within a transaction statement and may return an error;
```

File: examples/transaction-statement/transaction_statement.md [hunk 11]
```diff
@@ -1,10 +1,6 @@
 # Transaction statement
 
-Ballerina provides support for interacting with a transaction manager.
-Compile-time guarantees that transactions are bracketed with begin and `commit` or `rollback`.
-The region in the middle is typed as being a transactional context.
-Ballerina does not have a transactional memory and includes a transaction manager.
-The current transaction is a part of the execution context of a strand.
+Ballerina provides support for interacting with a transaction manager. Compile-time guarantees that transactions are bracketed with begin and `commit` or `rollback`. The region in the middle is typed as being a transactional context. Ballerina does not have a transactional memory and includes a transaction manager. The current transaction is a part of the execution context of a strand.
```
Comment (praneesha) [inline examples/transaction-statement/transaction_statement.md:3]:
```suggestion
Ballerina provides support for interacting with a transaction manager. The compile-time guarantees that transactions are bracketed with begin and `commit` or `rollback`. The region in the middle is typed as being a transactional context. Ballerina does not have a transactional memory and includes a transaction manager. The current transaction is a part of the execution context of a strand.
```

File: examples/transactional-qualifier/transactional_qualifier.bal [hunk 12]
```diff
@@ -16,18 +16,17 @@ public function main() returns error? {
...
-    // Calls the `bar()` transactional function.
+    // Calls `bar()` transactional function.
     bar(u);
     return;
 }
 
 function foo(Update u) {
     if transactional {
-        // This is a transactional context.
+        // This is transactional context.
```
Comment (praneesha) [inline examples/transactional-qualifier/transactional_qualifier.bal:28]:
```suggestion
        // This is the transactional context.
```

File: examples/transactional-qualifier/transactional_qualifier.md [hunk 13]
```diff
@@ -1,7 +1,6 @@
 # Transactional qualifier
 
-At compile-time, regions of code are typed as being a transactional context.
-Ballerina guarantees that, whenever that region is executed, there will be a current transaction.
+At compile-time, for regions of code are typed as being a transactional context; Ballerina guarantees that, whenever that region is executed, there will be a current transaction.
```
Comment (praneesha) [inline examples/transactional-qualifier/transactional_qualifier.md:3]:
```suggestion
At compile-time, regions of code are typed as being transactional context; Ballerina guarantees that whenever that region is executed, there will be a current transaction.
```

**Review by KRVPerera — COMMENTED** [review]
Missing `$` sign for `rollback` and `transactional_named_workers` BBEs

File: examples/retry-transaction-statement/retry_transaction_statement.bal (hunk 5 above)
Comment (pcnfernando) [reply examples/retry-transaction-statement/retry_transaction_statement.bal:23]: This should be 
```suggestion
    // Returns an `error:Retriable` error for retrying.
```

File: examples/transactional-qualifier/transactional_qualifier.bal (hunk 12 above)
Comment (pcnfernando) [reply examples/transactional-qualifier/transactional_qualifier.bal:28]: This should be 
```suggestion
        // This is in the transactional context.
```

File: examples/retry-transaction-statement/retry_transaction_statement.md (hunk 8 above)
Comment (pcnfernando) [reply examples/retry-transaction-statement/retry_transaction_statement.md]: Updated in 48f771040254c32ba752611f95d753cbb6dfa925

File: examples/retry-transaction-statement/retry_transaction_statement.md [hunk 14]
```diff
@@ -1,10 +1,10 @@
 # Retry transaction statement
 
 Transactional errors are often transient: retrying will fix them.
-This works by creating a  RetryManager object `r`, before executing the transaction.
-If the block fails with error `e`, it calls `r.shouldRetry(e)`.
-If that returns `true`, then it executes the block again.
-`retry` has an optional type parameter giving class of `RetryManager` to create, and optional arguments to new `DefaultRetryManager` tries `n` times.
+This works by creating a `RetryManager` object `r` before executing the transaction.
+If the block fails with an error `e`, it calls `r.shouldRetry(e)`. If that returns `true`, then it executes the block again.
+`retry` statement accepts an optional type parameter which is a class of `RetryManager` and optional arguments for the class's init method.
```
Comment (praneesha) [inline examples/retry-transaction-statement/retry_transaction_statement.md:6]:
```suggestion
The `retry` statement accepts an optional type parameter, which is a class of the `RetryManager` and optional arguments for the `init` method of the class.
```

File: examples/transactional-qualifier/transactional_qualifier.md (hunk 13 above)
Comment (pcnfernando) [reply examples/transactional-qualifier/transactional_qualifier.md]:
```suggestion
At compile-time, regions of code are typed as being transactional context; Ballerina guarantees that, whenever that region is executed, there will be a current transaction.
```

File: examples/transactional-qualifier/transactional_qualifier.md [hunk 15]
```diff
@@ -1,7 +1,6 @@
 # Transactional qualifier
 
-At compile-time, regions of code are typed as being a transactional context.
-Ballerina guarantees that, whenever that region is executed, there will be a current transaction.
+At compile-time, regions of code are typed as being transactional context; Ballerina guarantees that whenever that region is executed, there will be a current transaction.
```
Comment (pcnfernando) [inline examples/transactional-qualifier/transactional_qualifier.md:3]:
```suggestion
At compile-time, regions of code are typed as being transactional context; Ballerina guarantees that whenever that region is executed, there will be a current transaction.
```

**Comment by pcnfernando** [comment]
> Missing `$` sign for `rollback` and `transactional_named_workers` BBEs

This is done

**Review by praneesha — APPROVED** [review]

**Pull request merged**